powertools-parameters/src/main/java/software/amazon/lambda/powertools/parameters/ParamManager.java

```diff
@@ -27,22 +32,39 @@
     private static final CacheManager cacheManager = new CacheManager();
     private static final TransformationManager transformationManager = new TransformationManager();
 
-    private static SecretsProvider secretsProvider;
-    private static SSMProvider ssmProvider;
+    private static ConcurrentHashMap<Class<? extends BaseProvider>, BaseProvider> providers = new ConcurrentHashMap<>();
+
+    /**
+     * Get a concrete implementation of {@link BaseProvider}.<br/>
+     * You can specify {@link SecretsProvider} or {@link SSMProvider} or create your custom provider
+     * by extending {@link BaseProvider} if you need to integrate with a different parameter store.
+     * @return a {@link SecretsProvider}
+     */
+    public static <T extends BaseProvider> T getProvider(Class<T> providerClass) {
+        if (providerClass == null) {
+            throw new IllegalStateException("You cannot provide a null provider class.");
+        }
+        try {
+
+            if(!providers.containsKey(providerClass)) {
+                Constructor<T> constructor = providerClass.getDeclaredConstructor(CacheManager.class);
+                T provider = constructor.newInstance(cacheManager);
+                provider.setTransformationManager(transformationManager);
+                providers.putIfAbsent(providerClass, provider);
+            }
+            return (T) providers.get(providerClass);
+        } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+            throw new ProviderException(e);
```

Comment:
Since this exception is thrown out to clients if at all if occurs, we should probably have better error message for clients to guide them ? [For example](https://github.com/awslabs/aws-lambda-powertools-java/blob/master/powertools-sqs/src/main/java/software/amazon/lambda/powertools/sqs/SqsUtils.java#L260)